modules/networking/main.tf

```diff
@@ -34,3 +34,54 @@ resource "azurerm_subnet" "redis" {
   address_prefixes     = [var.network_redis_subnet_cidr]
   virtual_network_name = azurerm_virtual_network.default.name
 }
+
+resource "azurerm_network_security_group" "nsg" {
+  count               = length(var.allowed_ip_ranges) > 0 ? 1 : 0
+  name                = "${var.namespace}-allowlist-nsg"
+  location            = var.location
+  resource_group_name = var.resource_group_name
+  tags                = var.tags
+}
+
+
+resource "azurerm_network_security_rule" "this" {
```

Comment:
we use `default` instead of this, small difference all the other best practices are the same

Reply:
resource "azurerm_network_security_rule" "Default" {
  count                       = length(var.allowed_ip_ranges) > 0 ? length(var.allowed_ip_ranges) : 0
  name                        = "allowRule-${count.index}"
  priority                    = 100 + "${count.index}"
  direction                   = "Inbound"
  access                      = "Allow"
  protocol                    = "Tcp"
  source_port_range           = "*"
  destination_port_range      = "*"
  source_address_prefixes     = [var.allowed_ip_ranges[count.index]]
  destination_address_prefix  = "*"
  resource_group_name         = var.resource_group_name
  network_security_group_name = azurerm_network_security_group.default.0.name
  depends_on                  = [azurerm_network_security_group.default]
}



resource "azurerm_network_security_rule" "default" {
  count                       = length(var.allowed_ip_ranges) > 0 ? 1 : 0
  name                        = "defaultAppGatewayV2SkuRule"
  priority                    = 120
  direction                   = "Inbound"
  access                      = "Allow"
  protocol                    = "*"
  source_port_range           = "*"
  destination_port_range      = "65200-65535"
  source_address_prefix       = "*"
  destination_address_prefix  = "*"
  resource_group_name         = var.resource_group_name
  network_security_group_name = azurerm_network_security_group.default.0.name
}